File: app/src/main/java/com/ivanovsky/passnotes/util/ReflectionUtils.java
package com.ivanovsky.passnotes.util;

import com.annimon.stream.Stream;

public class ReflectionUtils {

	public static boolean containsInterfaceInClass(Class classType, Class interfaceType) {
		return Stream.of(classType.getInterfaces())
				.anyMatch(classInterface -> classInterface == interfaceType);
	}
}

File: app/src/main/kotlin/com/ivanovsky/passnotes/util/ReflectionUtils.kt
package com.ivanovsky.passnotes.util

object ReflectionUtils {

    @JvmStatic
    fun containsInterfaceInClass(classType: Class<*>, interfaceType: Class<*>): Boolean {
        return classType.interfaces
            .any { classInterface -> classInterface == interfaceType }
    }

    fun getClassByName(typeName: String): Class<*>? {
        return try {
            Class.forName(typeName)
        } catch (e: ClassNotFoundException) {
            null
        }
    }
}